libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -925,6 +1070,102 @@ impl NodeAPI for Greenlight {
         }
     }
 
+    async fn max_amount_to_send(
```

Comment:
This method assumes all channels are with the same peer. You'll want to either get a path for each peer, or filter the used channels to that single peer. Because a different peer will have a different route and different fees.

Reply:
Good point! we probably need to filter only these channels that reaches the first hop of the route

Reply:
Changed to only consider channels for the first hop.